extensions/html-language-features/server/src/modes/javascriptMode.ts

```diff
@@ -35,13 +44,17 @@ function getLanguageServiceHost(scriptKind: ts.ScriptKind) {
 			getScriptVersion: (fileName: string) => {
 				if (fileName === currentTextDocument.uri) {
 					return String(currentTextDocument.version);
+				} else if (isFileURI(fileName) && ts.sys.fileExists(uriToFilePath(fileName))) {
+					return String(ts.sys.getModifiedTime!(uriToFilePath(fileName)) ?? '1');
 				}
 				return '1'; // default lib an jquery.d.ts are static
 			},
 			getScriptSnapshot: (fileName: string) => {
 				let text = '';
 				if (fileName === currentTextDocument.uri) {
 					text = currentTextDocument.getText();
+				} else if (isFileURI(fileName) && ts.sys.fileExists(uriToFilePath(fileName))) {
```

Comment:
If the requested file is on a local file system, read it with `ts.sys.readFile`. (Web IDE support is not possible since TS requires sync access, as discussed in https://github.com/microsoft/vscode/pull/121517)